### PR TITLE
Project refactor

### PR DESCRIPTION
**Pull request by floydprice**

### CODE REVIEW

**Comment by floydprice** [comment]
@dashton82 turned this in to a PR just to make our collaboration more awesome

File: package.json [hunk 1]
```diff
@@ -1,5 +1,11 @@
 {
   "name": "login.dfe.directories",
+  "jest": {
+    "testEnvironment": "node",
+    "collectCoverage": false,
+    "clearMocks": true,
+    "testRegex": ".*\\.test.js$"
```
Comment (simonbu11) [inline package.json]: Now there is not a split of jest and mocha, we do not need this

File: package.json [hunk 2]
```diff
@@ -23,6 +29,8 @@
     "eslint-config-airbnb-base": "^11.3.1",
     "eslint-plugin-import": "^2.7.0",
     "ioredis-mock": "^2.4.0",
+    "jest": "^21.2.1",
+    "jest-cli": "^21.2.1",
     "mocha": "^3.5.3",
```
Comment (simonbu11) [inline package.json]: This can be removed now

File: src/app/userCodes/data/redisUserCodeStorage.js [hunk 3]
```diff
@@ -39,39 +39,38 @@ const deleteCode = async (uid, client) => {
...
     }
   }
 
   async getUserPasswordResetCode(uid) {
     try {
-      return await find(uid, redisClient);
+      return await find(uid, this.client);
     } catch (e) {
-      throw e;
+      logger.error(e);
```
Comment (simonbu11) [inline src/app/userCodes/data/redisUserCodeStorage.js]: Should we be throwing the error here? If not error it possibly not an appropriate log level

File: src/app/userCodes/data/redisUserCodeStorage.js [hunk 4]
```diff
@@ -39,39 +39,38 @@ const deleteCode = async (uid, client) => {
...
     }
   }
 
   async createUserPasswordResetCode(uid, clientId) {
     try {
-      return await createCode(uid, clientId, redisClient);
+      return await createCode(uid, clientId, this.client);
     } catch (e) {
-      throw e;
+      logger.error(e);
```
Comment (simonbu11) [inline src/app/userCodes/data/redisUserCodeStorage.js]: Should we be throwing the error here? If not error it possibly not an appropriate log level

File: src/app/userCodes/data/redisUserCodeStorage.js [hunk 5]
```diff
@@ -39,39 +39,38 @@ const deleteCode = async (uid, client) => {
...
     }
   }
 
   async deleteUserPasswordResetCode(uid) {
     try {
-      return await deleteCode(uid, redisClient);
+      return await deleteCode(uid, this.client);
     } catch (e) {
-      throw e;
+      logger.error(e);
```
Comment (simonbu11) [inline src/app/userCodes/data/redisUserCodeStorage.js]: Should we be throwing the error here? If not error it possibly not an appropriate log level

File: test/userCodesTests/delete.test.js [hunk 6]
```diff
@@ -1,25 +1,13 @@
-const expect = require('chai').expect;
-const deleteUserCode = require('./../../src/userCodes/delete');
+jest.mock('./../../src/app/userCodes/data/redisUserCodeStorage');
+const deleteUserCode = require('../../src/app/userCodes/api/delete');
```
Comment (simonbu11) [inline test/userCodesTests/delete.test.js]: missing the ./ at beginning of require path

File: test/userCodesTests/redisUserCodeStorage.test.js [hunk 7]
```diff
@@ -1,131 +1,110 @@
...
+      return {
+        userCodes: {staticCode: false}
+      };
     });
+
+    config = require('./../../src/infrastructure/config');
+    config.mockImplementation(configStub);
+
+    redis = new RedisMock();
+    const userCodeStorage = require('../../src/app/userCodes/data/redisUserCodeStorage');
```
Comment (simonbu11) [inline test/userCodesTests/redisUserCodeStorage.test.js]: Missing ./ at beginning of require path

File: test/userCodesTests/validate.test.js [hunk 8]
```diff
@@ -29,6 +19,16 @@ describe('When validating a user code', () => {
         code: 'ABC123'
       }
     };
+
+    getUserStub = jest.fn().mockImplementation(()=>{ return new Promise((resolve)=>{      resolve(getResponse);    })});
```
Comment (simonbu11) [inline test/userCodesTests/validate.test.js]: I think the style is that lambda should be on multiple lines

File: test/usertests/UserAzureActiveDirectoryAdapter.test.js [hunk 9]
```diff
@@ -1,120 +1,93 @@
...
+jest.mock('activedirectory');
+jest.mock('login.dfe.request-verification');
 
-const user = {
-  dn: 'test',
-  givenName: 'Test',
-  sn: 'Tester',
-  userPrincipalName: 'test@localuser.com'
-}; // '{"email": "test@localuser.com", "first_name": "Test", "last_name" : "Tester"}';
+const UserAzureActiveDirectoryAdapter = require('../../src/app/user/adapter/UserAzureActiveDirectoryAdapter');
```
Comment (simonbu11) [inline test/usertests/UserAzureActiveDirectoryAdapter.test.js]: Missing ./ at start of require path

File: test/usertests/UserRedisAdapter.test.js [hunk 10]
```diff
@@ -22,30 +20,30 @@ describe('When using redis storage service', () => {
 
       const actual = await userStorage.findByUsername('test@localuser.com');
 
-      expect(actual).to.not.equal(undefined);
-      expect(JSON.stringify(actual)).to.equal('{"sub":"test@localuser.com"}');
+      expect(actual).not.toBe(undefined);
```
Comment (simonbu11) [inline test/usertests/UserRedisAdapter.test.js]: Can use [.toBeUndefined()](https://facebook.github.io/jest/docs/en/expect.html#tobeundefined)

File: test/usertests/UserRedisAdapter.test.js [hunk 11]
```diff
@@ -22,30 +20,30 @@ describe('When using redis storage service', () => {
...
     });
     it('then the json is parsed and returned', async () => {
       redis.set('Users', '[{"sub": "54321", "email":"test4@localuser.com"},{"sub": "12345", "email":"test3@localuser.com"}]');
       redis.set('User_12345', '{"sub": "test3@localuser.com","email":"test3@localuser.com", "first_name": "Tester", "last_name" : "Testing"}');
 
       const actual = await userStorage.findByUsername('test3@localuser.com');
 
-      expect(actual).to.not.equal(null);
-      expect(actual.first_name).to.equal('Tester');
+      expect(actual).not.toBe(null);
```
Comment (simonbu11) [inline test/usertests/UserRedisAdapter.test.js]: can use .toBeNull()

File: test/usertests/UserRedisAdapter.test.js [hunk 12]
```diff
@@ -22,30 +20,30 @@ describe('When using redis storage service', () => {
...
+      expect(actual.first_name).toBe('Tester');
     });
     it('then if the user is not found then null is returned', async () => {
       redis.set('Users', '[{"sub": "54321", "email":"test4@localuser.com"},{"sub": "12345", "email":"test3@localuser.com"}]');
       redis.set('User_12345', '{"sub": "test3@localuser.com","email":"test3@localuser.com", "first_name": "Tester", "last_name" : "Testing"}');
 
       const actual = await userStorage.findByUsername('test4@localuser.com');
 
-      expect(actual).to.equal(null);
+      expect(actual).toBe(null);
```
Comment (simonbu11) [inline test/usertests/UserRedisAdapter.test.js]: can use .toBeNull()

File: test/usertests/UserRedisAdapter.test.js [hunk 13]
```diff
@@ -62,20 +60,20 @@ describe('When using redis storage service', () => {
...
     it('if the user exists the record is updated', async () => {
       redis.set('User_test3@localuser.com', '{"sub": "test3@localuser.com","email":"test3@localuser.com", "first_name": "Tester", "last_name" : "Testing", "salt":"123456768"}');
 
       const actual = await userStorage.changePassword('test3@localuser.com', 'my-new-password');
 
-      expect(actual).to.equal(true);
+      expect(actual).toBe(true);
       const findResult = await userStorage.find('test3@localuser.com');
-      expect(findResult).to.not.equal(null);
+      expect(findResult).not.toBe(null);
```
Comment (simonbu11) [inline test/usertests/UserRedisAdapter.test.js]: can use .toBeNull()

File: package.json (hunk 2 above)
Comment (dashton82) [reply package.json]: Also removed proxyquire and sinon

**Pull request merged**